src/App.tsx

```diff
@@ -15,48 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabeticaly = 'alphabeticaly',
+  Length = 'length',
+}
+
+interface SortParams {
+  sortField: string;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+) {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabeticaly:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const sortedGoods = getPreparedGoods(
+    goodsFromServer, { sortField, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortField !== SortType.Alphabeticaly },
+          )}
+          onClick={() => setSortField(SortType.Alphabeticaly)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortField !== SortType.Length },
+          )}
+          onClick={() => setSortField(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || isReversed) && (
+          <button
```

Comment:
I really don`t get why, but your reset button goes fully red if you click only on reverse and it does not fully match the working version. Also it would be cool to create seperate function for reset button as we could have more things to reset.
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/100444192/48a3abd3-b3c8-4363-9ba6-c86041cb8712)